### PR TITLE
feat: support tachyon halo2

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR enables producing a proof by Tachyon.

### CODE REVIEW

File: halo2_proofs/src/plonk/tachyon.rs [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+    xor_shift_rng::XORShiftRng as TachyonXORShiftRng,
+};
+use ff::Field;
+use halo2curves::{
+    bn256::{Fr, G1Affine, G1},
+    group::{prime::PrimeCurveAffine, Curve},
+    CurveAffine,
+};
+
+/// This creates a proof for the provided `circuit` when given the public
```
Comment (fakedev9999) [inline halo2_proofs/src/plonk/tachyon.rs]:
```suggestion
/// This creates a proof for the provided `circuits` when given the public
```

File: halo2_proofs/src/plonk/tachyon.rs [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+                    advice_assigned_values.as_slice(),
+                    advice_values.as_mut_slice(),
+                );
+
+                // Add blinding factors to advice columns
+                for advice_values in &mut advice_values {
+                    //for cell in &mut advice_values[unusable_rows_start..] {
+                    //*cell = C::Scalar::random(&mut rng);
+                    //*cell = C::Scalar::one();
+                    //}
```
Comment (fakedev9999) [inline halo2_proofs/src/plonk/tachyon.rs]:
```suggestion
```

File: halo2_proofs/src/plonk/prover.rs [hunk 3]
```diff
@@ -502,11 +507,13 @@ pub fn create_proof<
...
     // Commit to the vanishing argument's random polynomial for blinding h(x_3)
     let vanishing = vanishing::Argument::commit(params, domain, &mut rng, transcript)?;
+    println!("beta-gamma-elapsed-time: {:#?}", start.elapsed());
 
+    println!("<YPhase>");
+    let start = Instant::now();
     // Obtain challenge for keeping all separate gates linearly independent
     let y: ChallengeY<_> = transcript.squeeze_challenge_scalar();
     println!("Halo2(y): {:?}", *y);
-
```
Comment (fakedev9999) [inline halo2_proofs/src/plonk/prover.rs:505]:
```suggetion

```

**Review by fakedev9999 — CHANGES_REQUESTED** [review]

File: halo2_proofs/src/plonk/tachyon.rs (hunk 1 above)
Comment (chokobole) [reply halo2_proofs/src/plonk/tachyon.rs:28]: Actually, since I just took this comment from Halo2, I'm worried his may produce another diffs !

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Review by fakedev9999 — APPROVED** [review]
LGTM

**Pull request merged**